packages/graphile-build-pg/res/introspection-query.sql

```diff
@@ -104,13 +104,6 @@ with
       left join pg_catalog.pg_description as dsc on dsc.objoid = rel.oid and dsc.objsubid = 0
       left join pg_catalog.pg_namespace as nsp on nsp.oid = rel.relnamespace
     where
-      -- Select classes that are in our namespace, or are referenced in a
-      -- procedure.
-      (
-        rel.relnamespace in (select "id" from namespace) or
-        rel.reltype in (select "returnTypeId" from procedure) or
-        rel.reltype in (select unnest("argTypeIds") from procedure)
-      ) and
```

Comment:
This was the crux of the issue, making this filter broad enough was a real challenge. Instead I removed it and perform on-demand type definition rather than eagerly creating all the classes.